src/modules/twap/hooks/useTwapOrdersExecutions.ts

```diff
@@ -0,0 +1,48 @@
+import { useAtomValue } from 'jotai'
+import { useMemo } from 'react'
+
+import { Order, OrderInfoApi } from 'legacy/state/orders/actions'
+import { useOrders } from 'legacy/state/orders/hooks'
+
+import { useWalletInfo } from 'modules/wallet'
+
+import { DEFAULT_TWAP_EXECUTION_INFO } from '../const'
+import { twapPartOrdersAtom } from '../state/twapPartOrdersAtom'
+import { TwapOrderExecutionInfo } from '../types'
+
+export type TwapOrdersExecutionMap = { [id: string]: TwapOrderExecutionInfo }
+
+export function useTwapOrdersExecutions(ids: string[]): TwapOrdersExecutionMap {
+  const { chainId } = useWalletInfo()
+  const twapPartOrders = useAtomValue(twapPartOrdersAtom)
+
+  const allOrders = useOrders({ chainId })
+
+  return useMemo(() => {
+    return ids.reduce<TwapOrdersExecutionMap>((acc, id) => {
+      const childrenIds = (twapPartOrders[id] || []).map((item) => item.uid)
+
+      if (childrenIds.length > 0) {
+        const children = allOrders.filter((order) => childrenIds.includes(order.id))
```

Comment:
How often we see these ids changing?

Because this feels highly inefficient:
- For every id go over the list of item to get children ids
- For every order, check whether it's in the children ids list

Could we:
- Have a Set with children ids to check each order id rather than always looping through all of them.
OR
- Have a Map with all orders so we can search a (maybe smaller subset) looping over children ids and checking for an existing order for each?

Reply:
Great point! Thanks!
Refactored it to make O(n)